### PR TITLE
feature(core): performance - move fetchTokens to WalletDataProvider to prevent multi…

### PR DESCRIPTION
**Pull request by thedoublejay**
…ple re-renders #2242 (#2420)

Co-authored-by: JJ Adonis <josejaimecareers@gmail.com>

#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes https://github.com/DeFiCh/wallet/issues/2242

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/AppNavigator.tsx [hunk 1]
```diff
@@ -20,6 +27,17 @@ export interface AppParamList {
...
   const { isLight } = useThemeContext()
   const DeFiChainTheme: Theme = getDefaultTheme(isLight)
+  const blockCount = useSelector((state: RootState) => state.block.count)
+  const { address } = useWalletContext()
+  const client = useWhaleApiClient()
+  const { network } = useNetworkContext()
+  const dispatch = useDispatch()
+
+  /* Global polling based on blockCount, network and address */
+  useEffect(() => {
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/AppNavigator.tsx:37]: I remembered we have `WalletDataProvider.tsx` for global polling code, to fetch pool pairs and dex price, why not place it there too?

Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/AppNavigator.tsx:37]: @kyleleow It's because of @pierregee PR - https://github.com/DeFiCh/wallet/pull/2718

Comment (kyleleow) [reply mobile-app/app/screens/AppNavigator/AppNavigator.tsx:37]: My suggestion is to move `fetchTokens` to `<WalletDataProvider>`, and move `<WalletDataProvider>` to `AppNavigator` from `App.tsx`. 

So we can still keep the global polling code at one file. What do you think?

Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/AppNavigator.tsx:37]: We can. Haven't explored that option yet. What do you think @pierregee ?

Comment (pierregee) [reply mobile-app/app/screens/AppNavigator/AppNavigator.tsx:37]: Yea I agree to keep the maintanability, didn't explore that yet too. I'll have a look!

Comment (kyleleow) [reply mobile-app/app/screens/AppNavigator/AppNavigator.tsx:37]: If it doesnt work, i would suggest plan B:
- create another file called `WalletAddressDataProvider` to store global polling code that specifically requires `WalletContext`

**Review by fullstackninja864 — APPROVED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx [hunk 2]
```diff
@@ -68,16 +65,9 @@ export function PlaceBidScreen (props: Props): JSX.Element {
   const { blocksRemaining } = useAuctionTime(vault.liquidationHeight, blockCount)
   const logger = useLogger()
   const client = useWhaleApiClient()
-  const isFocused = useIsFocused()
 
   const [bidAmount, setBidAmount] = useState<string>('')
 
-  useEffect(() => {
-    if (isFocused) {
```
[the remark this answers is not in the document]
Comment (fullstackninja864) [reply mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx:76]: We use fetchTokens at many places, so global poling will definitely be beneficial. I guess the above implementation will slightly improve performance.

**Pull request merged**